pennylane_lightning/lightning_qubit.py

```diff
@@ -295,13 +304,31 @@ def adjoint_jacobian(self, tape, starting_state=None, use_device_state=False):
 
         state_vector = StateVectorC64(ket) if use_csingle else StateVectorC128(ket)
 
-        jac = adj.adjoint_jacobian(
-            state_vector,
-            obs_serialized,
-            ops_serialized,
-            tp_shift,
-            tape.num_params,
-        )
+        # If requested batching over observables, chunk into OMP_NUM_THREADS sized chunks.
+        # This will allow use of Lightning with adjoint for large-qubit numbers AND large
+        # numbers of observables, enabling choice between compute time and memory use.
+        if self._batch_obs:
+            requested_threads = int(os.getenv("OMP_NUM_THREADS", "1"))
```

Comment:
Wouldn't be a good idea to get `requested_threads` before `if` and so check if `requested_threads==1` jump on `else` instead of the call to `_chunk_iterable`? :thinking:

Reply:
Hmmm, good point. Something like:
```
if self._batch_obs and requested_threads > 1:
```
?

Reply:
Yes! something like this.